### PR TITLE
Add GKE DPv2 observability dashboards

### PR DESCRIPTION
**Pull request by siwiutki**
Tracking issue: b/325017847

### CODE REVIEW

File: dashboards/google-kubernetes-engine/metadata.yaml [hunk 1]
```diff
@@ -60,3 +60,13 @@ sample_dashboards:
     id: gke-optimization-signals
     display_name: GKE Optimization Signals
     description: "This dashboard has 10 charts using the related [Kubernetes metrics](https://cloud.google.com/monitoring/api/metrics_kubernetes) and summarize GKE cost optimization signals based on the [State of Kubernetes Cost Optimization](https://cloud.google.com/blog/products/containers-kubernetes/new-report-state-of-kubernetes-cost-optimization) report includes demand scaling, spot instance usage, cluster bin packing and workload rightsizing."
+  -
+    category: Google Kubernetes Engine
+    id: gke-dpv2-observability-cluster-flows-overview
+    display_name: GKE DPv2 Observability - Cluster Flows Overview
+    description: "This dashboard features 6 charts with overview data about the pod-to-workload traffic in your GKE cluster. Monitor how much of your traffic is dropped, see what are the most common drop reasons, all at a glance. [Dataplane V2 metrics must be enabled](https://cloud.devsite.corp.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard."
```
Comment (marqc) [inline dashboards/google-kubernetes-engine/metadata.yaml:67]:
```suggestion
    description: "This dashboard features 6 charts with overview data about the pod-to-workload traffic in your GKE cluster. Monitor how much of your traffic is dropped, see what are the most common drop reasons, all at a glance. [Dataplane V2 metrics must be enabled](https://cloud.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard."
```

File: dashboards/google-kubernetes-engine/metadata.yaml [hunk 2]
```diff
@@ -60,3 +60,13 @@ sample_dashboards:
...
+  -
+    category: Google Kubernetes Engine
+    id: gke-dpv2-observability-drilldown-view
+    display_name: GKE DPv2 Observability - Drilldown View
+    description: "This dashboard features 4 charts providing pod-to-workload traffic data that is groupable and filterable, letting you gain more detailed insight into the traffic within your GKE cluster. Understand which workloads generate the most traffic, drill down which pods drop traffic and what's the reason, and more. [Dataplane V2 metrics must be enabled](https://cloud.devsite.corp.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard."
```
Comment (marqc) [inline dashboards/google-kubernetes-engine/metadata.yaml:72]:
```suggestion
    description: "This dashboard features 4 charts providing pod-to-workload traffic data that is groupable and filterable, letting you gain more detailed insight into the traffic within your GKE cluster. Understand which workloads generate the most traffic, drill down which pods drop traffic and what's the reason, and more. [Dataplane V2 metrics must be enabled](https://cloud.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard."
```

File: dashboards/google-kubernetes-engine/README.md [hunk 3]
```diff
@@ -54,4 +54,18 @@
...
+|:---------------------|
+|Filename: [gke-dpv2-observability-cluster-flows-overview.json](gke-dpv2-observability-cluster-flows-overview.json)|
+|This dashboard features 6 charts with overview data about the pod-to-workload traffic in your GKE cluster. Monitor how much of your traffic is dropped, see what are the most common drop reasons, all at a glance. [Dataplane V2 metrics must be enabled](https://cloud.devsite.corp.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard.|
+
+&nbsp;
+
+|GKE DPv2 Observability - Drilldown View|
+|:---------------------|
+|Filename: [gke-dpv2-observability-drilldown-view.json](gke-dpv2-observability-drilldown-view.json)|
+|This dashboard features 4 charts providing pod-to-workload traffic data that is groupable and filterable, letting you gain more detailed insight into the traffic within your GKE cluster. Understand which workloads generate the most traffic, drill down which pods drop traffic and what's the reason, and more. [Dataplane V2 metrics must be enabled](https://cloud.devsite.corp.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard.|
```
Comment (marqc) [inline dashboards/google-kubernetes-engine/README.md:69]:
```suggestion
|This dashboard features 4 charts providing pod-to-workload traffic data that is groupable and filterable, letting you gain more detailed insight into the traffic within your GKE cluster. Understand which workloads generate the most traffic, drill down which pods drop traffic and what's the reason, and more. [Dataplane V2 metrics must be enabled](https://cloud.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard.|
```

File: dashboards/google-kubernetes-engine/README.md [hunk 4]
```diff
@@ -54,4 +54,18 @@
 |Filename: [gke-optimization-signals.json](gke-optimization-signals.json)|
 |This dashboard has 10 charts using the related [Kubernetes metrics](https://cloud.google.com/monitoring/api/metrics_kubernetes) and summarize GKE cost optimization signals based on the [State of Kubernetes Cost Optimization](https://cloud.google.com/blog/products/containers-kubernetes/new-report-state-of-kubernetes-cost-optimization) report includes demand scaling, spot instance usage, cluster bin packing and workload rightsizing.|
 
+&nbsp;
+
+|GKE DPv2 Observability - Cluster Flows Overview|
+|:---------------------|
+|Filename: [gke-dpv2-observability-cluster-flows-overview.json](gke-dpv2-observability-cluster-flows-overview.json)|
+|This dashboard features 6 charts with overview data about the pod-to-workload traffic in your GKE cluster. Monitor how much of your traffic is dropped, see what are the most common drop reasons, all at a glance. [Dataplane V2 metrics must be enabled](https://cloud.devsite.corp.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard.|
```
Comment (marqc) [inline dashboards/google-kubernetes-engine/README.md:62]:
```suggestion
|This dashboard features 6 charts with overview data about the pod-to-workload traffic in your GKE cluster. Monitor how much of your traffic is dropped, see what are the most common drop reasons, all at a glance. [Dataplane V2 metrics must be enabled](https://cloud.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability) to make use of this dashboard.|
```

File: dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+      },
+      {
+        "xPos": 0,
+        "yPos": 0,
+        "width": 48,
+        "height": 6,
+        "widget": {
+          "title": "Meaning of \"flow\"",
+          "text": {
+            "content": "A _flow_ is generated and measured when the eBPF program observes a network packet with a given tuple of (source ip, source port, destination ip, destination port, tcp flags, protocol) and there was no event with exactly the same tuple of these properties within the last five seconds.",
```
Comment (marqc) [inline dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json:319]:
```suggestion
            "content": "A _flow_ is generated and measured when the eBPF program (on each node) observes a network packet with a given tuple of (source ip, source port, destination ip, destination port, tcp flags, protocol) and there was no event with exactly the same tuple of these properties within the last five seconds.",
```

File: dashboards/google-kubernetes-engine/gke-dpv2-observability-drilldown-view.json [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+      },
+      {
+        "xPos": 0,
+        "yPos": 10,
+        "width": 48,
+        "height": 6,
+        "widget": {
+          "title": "Meaning of \"flow\"",
+          "text": {
+            "content": "A _flow_ is generated and measured when the eBPF program observes a network packet with a given tuple of (source ip, source port, destination ip, destination port, tcp flags, protocol) and there was no event with exactly the same tuple of these properties within the last five seconds.",
```
Comment (marqc) [inline dashboards/google-kubernetes-engine/gke-dpv2-observability-drilldown-view.json:277]:
```suggestion
            "content": "A _flow_ is generated and measured when the eBPF program (on each node) observes a network packet with a given tuple of (source ip, source port, destination ip, destination port, tcp flags, protocol) and there was no event with exactly the same tuple of these properties within the last five seconds.",
```

**Review by marqc — CHANGES_REQUESTED** [review]

**Review by marqc — APPROVED** [review]
lgtm

File: dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json [hunk 7]
```diff
@@ -0,0 +1,336 @@
+{
+  "displayName": "GKE DPv2 Observability - Cluster Flows Overview",
+  "dashboardFilters": [],
+  "category": "CUSTOM",
+  "etag": "51a3c4a0f285f7738a6d04662347d927",
```
Comment (pintofloyed) [inline dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json:5]: The etag needs to be removed

Comment (siwiutki) [reply dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json]: Done.

File: dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+              "padding": "P_EXTRA_SMALL",
+              "pointerLocation": "POINTER_LOCATION_UNSPECIFIED",
+              "textColor": "#212121",
+              "verticalAlignment": "V_TOP"
+            }
+          }
+        }
+      }
+    ],
+    "legacyRows": 58
```
[the remark this answers is not in the document]
Comment (siwiutki) [reply dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json]: Sure, removed these fields

File: dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+              "horizontalAlignment": "H_LEFT",
+              "padding": "P_EXTRA_SMALL",
+              "pointerLocation": "POINTER_LOCATION_UNSPECIFIED",
+              "textColor": "#212121",
+              "verticalAlignment": "V_TOP"
+            }
+          }
+        }
+      }
+    ],
```
Comment (pintofloyed) [inline dashboards/google-kubernetes-engine/gke-dpv2-observability-cluster-flows-overview.json:320]: These trailing commas on line 320 and 321 are causing errors in the template. Similar case with the other file

**Review by pintofloyed — APPROVED** [review]

**Pull request merged**